.github/workflows/create-release-main.yml

```diff
@@ -0,0 +1,71 @@
+name: Create new preview release (main)
+
+on:
+  # run when pushed to main is published,
+  # which creates a new tag
+  push:
+    branches:
+      - main
+
+env:
+  ARTIFACT_VERSION: experimental
+
+jobs:
+  build-and-push:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0 # required to access tags
+          submodules: "true"
+
+      - name: Build & run devcontainer image
+        # this always builds a new image from scratch rather than from the build-devcontainer-image workflow output
+        # so that we pick up the latest versions of everything
+        # NB: if you update this also update live-validation.yml
+        id: devcontainer
+        run: |
+          docker build --tag devcontainer:latest .devcontainer
+          mkdir -p $HOME/.docker # in case it doesn't exist
+          container_id=$(docker create -w /workspace -v $GITHUB_WORKSPACE:/workspace -v /var/run/docker.sock:/var/run/docker.sock devcontainer:latest)
+          docker start "$container_id"
+          echo "container_id=$container_id" >> $GITHUB_ENV
+
+      - name: Build required release files
+        run: |
+          container_id=${{env.container_id}}
+          docker exec "$container_id" task VERSION=${{ env.ARTIFACT_VERSION }} LATEST_VERSION_TAG=${{ env.ARTIFACT_VERSION }} make-release-artifacts
```

Comment:
Do we need to generate artifacts here as well? if yes, not sure where do we publish them?

Reply:
They are attached to the release it pushes them to, that's enough for this scenario imo